blog/2024-11-mastering-effect-and-afterRenderEffect/README.md

```diff
@@ -0,0 +1,477 @@
+| 2. `write`            | Use this phase to **write** to the DOM. **Never** read from the DOM in this phase. |
+| 3. `mixedReadWrite`   | Use this phase to read from and write to the DOM simultaneously. **Never** use this phase if it is possible to divide the work among the other phases instead. |
+| 4. `read`             | Use this phase to **read** from the DOM. **Never** write to the DOM in this phase. |
+
+[According to the docs](https://next.angular.dev/api/core/afterRenderEffect), you should prefer using the `read` and `write` phases over the `earlyRead` and `mixedReadWrite` phases when possible, to avoid performance degradation.
+
+To summarise this, with `afterRenderEffect()` you can do the following in the four phases:
+1. **earlyRead**: Read DOM properties before writes.
+2. **write**: Execute DOM write operations.
+3. **mixedReadWrite**: Allows for combined reads and writes but should be used sparingly!
+4. **read**: Execute DOM reads after writes are completed.
+5. ...and you can pass on a value (or an object) from one phase to the next.
+
+As mentionend before, there is also a second signature of `afterRenderEffect()` that accepts a single callback. 
+This function registers an effect to run after rendering is complete, specifically during the `mixedReadWrite` phase.
+However, the Angular documentation recommends specifying an explicit phase for the effect whenever possible to avoid potential performance issues.
+Therefore, we won't cover this signature in our article, as its usage is not recommended.
+
+
+### Phases Only Run Again When "Dirty" Through Signal Dependencies
+
+When `afterRenderEffect()` is initially called, all registered phases execute once in sequence.
+However, for any phase to run again, it must be marked as "dirty" due to a change in signal dependencies. 
+This dependency-based system helps Angular optimize performance by preventing redundant executions.
+
+For a phase to be marked "dirty" and eligible to rerun, it must establish a dependency on a signal that changes. 
+If a phase does not track any signals, or if the tracked signals remain unchanged, the phase won’t be marked as dirty, and its code will not re-execute.
+
+There are two main ways to establish dependencies in `afterRenderEffect()`:
+
+1. **Tracking the Value of a Previous Phase’s Output**: 
+  Each phase can return a value to be passed as input to the next phase (except `earlyRead`, which has no previous phase). 
+  This value is wrapped in a signal, and if we then read that signal in the following phase, we create a dependency. 
+  It's important to understand that we must actually execute the signal’s value function because simply passing the signal around is insufficient to establish a dependency.
```

Comment:
```suggestion
  It's important to understand that we must actually execute the signal’s getter function because simply passing the signal around is insufficient to establish a dependency.
```